### PR TITLE
Initial Implementation of Hydrator++ using Redux

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds `HydratorPlusPlusLeftPanelStore` as a `Redux` store.
- Adds `HydratorPlusPlusActions` `action creator` to interact with `Redux` store asynchronously.
- Adds `ReduxThunk` for asyn communication with the store.
- Adds unit test for `HydratorPlusPlusLeftPanelStore`.
- Removes unstable (old) unit tests for `mySettings` & `mySocket` services.
##### TODO:
- Need to write unit tests for `HydratorPlusPlusActions` service
- Need to write more negative unit tests for `HydratorPlusPlusLeftPanelStore` service.

### CODE REVIEW

File: cdap-ui/test/unit/settings-test.js [hunk 1]
```diff
@@ -1,68 +1,68 @@
...
-
-  });
-
-
-
-});
\ No newline at end of file
+// /*
+//  * Copyright © 2015 Cask Data, Inc.
+//  *
```
Comment (elfenheart) [inline cdap-ui/test/unit/settings-test.js:3]: just remove the file?

Comment (ajainarayanan) [inline cdap-ui/test/unit/settings-test.js:3]: No. We still need to fix this test. `mySettings` could be tested. It should be simple validations.

**Comment by elfenheart** [comment]
Getting this error:

screen shot 2016-05-04 at 11 42 42 am

**Comment by ajainarayanan** [comment]
~~@elfenheart Did you do `npm install`? https://github.com/caskdata/cdap/blob/feature/ui-redux-hydrator%2B%2B/cdap-ui/package.json#L60~~
Alright updated the npm-shrinkwrap.json file.

**Comment by elfenheart** [comment]
@ajainarayanan yup, I did a cache clean and do a fresh install as well.

**Comment by elfenheart** [comment]
It seems like there is an issue when there is a state change while you are in studio. Getting JS error. To reproduce, try out changing Pipeline Type

**Comment by ajainarayanan** [comment]
@elfenheart Verified the issues you mentioned and fixed. Added nodes to the canvas. Created a pipeline. Published it and ran it. Switched pipelines. Imported a pipeline. Chose a template. Went through possible scenarios I could think of. Could you please take another pass?

**Comment by elfenheart** [comment]
On hold for plugin template display name / filtering

**Comment by elfenheart** [comment]
👍  LGTM

**Pull request merged**